frontendVfsUtils/src/main/java/com/virtuslab/gitmachete/frontend/vfs/utils/MacheteFileType.java

```diff
@@ -0,0 +1,42 @@
+package com.virtuslab.gitmachete.frontend.vfs.utils;
+
+import javax.swing.Icon;
+
+import com.intellij.openapi.fileTypes.LanguageFileType;
+import com.intellij.openapi.fileTypes.PlainTextLanguage;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class MacheteFileType extends LanguageFileType {
+  public static final MacheteFileType INSTANCE = new MacheteFileType();
+
+  public static final String FILE_NAME = "machete";
+
+  protected MacheteFileType() {
+    super(PlainTextLanguage.INSTANCE);
+  }
+
+  @Override
+  public boolean isReadOnly() {
+    return false;
+  }
+
+  @Override
+  public String getName() {
+    return "Machete";
```

Comment:
Hmm where's that name displayed?

Reply:
It is not to be displayed, it is rather keyish value - to identify. The same value must be used within `plugin.xml` file type declaration. I changed it to "MACHETE_FILE".